### ISSUE TITLE
Server Function taking struct with empty Vec fails to deserialize

### ISSUE THREAD

**Issue by smessmer** [leptos-rs/leptos#1741]
**Describe the bug**

If a server function takes a `MyData` struct that contains a `Vec`, and then that server function is called with an empty `Vec`, it fails deserialization saying the whole field is missing.

```rust
#[derive(Serialize, Deserialize)]
struct Entry {
  id: i32,
}

#[derive(Serialize, Deserialize)]
struct MyData {
   // ... other fields
   some_list: Vec<Entry>,
}

#[server]
fn my_server_fun(data: MyData) {
}

#[component]
fn MyComponent() -> impl IntoView {
  let call_server_fn = create_action(move |_: &()| { my_server_fun(MyData{some_list: vec![]}) });
  view! {
    <button on:click=move || call_server_fun.dispatch() >{"Call server fun"}</button>
  }
}
```

Fails with:
```
{"Deserialization":"missing field `some_list`"}
```

Looking at the data sent over the network, it seems that it uses Multipart Encoding and sends the parameters as:
```
data[some_list][0]: ...
data[some_list][1]: ...
```
and when the list is empty, it just doesn't contain any entries for it. This is likely the cause of the issue.

Changing the encoding of the server function to `Cbor` works, so it's specific to the `Url` encoding.

**Leptos Dependencies**

```toml
leptos = { version = "^0.5.0-rc1", features = ["nightly"] }
leptos_axum = { version = "^0.5.0-rc1", optional = true }
leptos_meta = { version = "^0.5.0-rc1", features = ["nightly"] }
leptos_router = { version = "^0.5.0-rc1", features = ["nightly"] }
```

**To Reproduce**
1. Call a server function as described above with an empty list
2. Observe failure

**Expected behavior**
Server function is correctly called and deserializes its arguments

**Comment by dpytaylo** [comment]
I have encountered with the same problem. A empty vector cause this error 😥

**Comment by ChristopherPerry6060** [comment]
Some encodings have no method for displaying an empty list of items. Annotate your struct definition with `#[serde(default)]. This will cause serde to assume a missing field is to be replaced with the `Default::default` implementation of it.

```rust
#[derive(Serialize, Deserialize)]
struct MyData {
   #[serde(default)]
   some_list: Vec<Entry>,
}
```

**Comment by gbj** [comment]
I missed the fact that this was a nested struct originally. Yes I think @ChristopherPerry6060 has the correct answer here: add `#[serde(default)]` on the struct field or the whole struct.

@dpytaylo You say you've encountered the same issue: Is that similarly on a nested struct like this?

(I say "nested" because the server functions arguments are, themselves, built into a struct.)

It's also possible this is `serde_qs` specific behavior that makes encoding and decoding not perfectly reversible. (`serde_qs` is what's used for this server function URL encoding.)

**Comment by gbj** [comment]
Yeah so ultimately it's just this known issue with `serde_qs`: https://github.com/samscott89/serde_qs/issues/78

I guess we could add an attribute to the server fn macro that passes through `#[serde(default)]`.

**Comment by gbj** [comment]
This is closed by #1762: You can add `#[server(default)]` to any argument to a server function to indicate that it should use the default value, if absent.
